src/components/listbox.tsx

```diff
@@ -0,0 +1,214 @@
+import {
+  Listbox,
+  ListboxButton,
+  ListboxOption,
+  ListboxOptions,
+} from "@headlessui/react";
+import { AnimatePresence, easeInOut, motion } from "framer-motion";
+import { useInView } from "react-intersection-observer";
+
+interface DropdownProps {
+  options: string[];
+  currentOption: string;
+  setCurrentOption: (option: string) => void;
+  defaultOption?: string;
+  defaultOptionText?: string;
+  hint?: string;
+}
+
+const ListBoxItem = ({
+  option,
+  index,
+  defaultOption,
+  defaultOptionText,
+  currentOption,
+  currentOptionIndex,
+}: {
+  option: string;
+  index: number;
+  defaultOption?: string;
+  defaultOptionText?: string;
+  currentOption: string;
+  currentOptionIndex: number;
+}) => {
+  const [ref, inView] = useInView({
+    triggerOnce: true,
+    threshold: 0.1,
+    // Ensure selected item is considered in view immediately
+    initialInView: option === currentOption,
+  });
+
+  const getStaggerDelay = () => {
+    const selectedIndex = currentOptionIndex;
+    const distanceFromSelected = Math.abs(index - selectedIndex);
+    if (selectedIndex < 8 && index < 8) {
+      return (index + 1) * 0.05;
+    } else if (index - selectedIndex > -10 && index - selectedIndex <= 0) {
+      return (10 - distanceFromSelected) * 0.05;
+    }
+    return 0.05;
+  };
+
+  const itemVariants = {
+    hidden: {
+      opacity: 0,
+      y: -10,
+    },
+    visible: {
+      opacity: 1,
+      y: 0,
+      transition: {
+        duration: 0.2,
+        ease: easeInOut,
+        delay: getStaggerDelay(),
+      },
+    },
+    exit: {
+      opacity: 0,
+      y: -5,
+      transition: {
+        duration: 0.2,
+        ease: easeInOut,
+      },
+    },
+  };
+
+  return (
+    <ListboxOption
+      key={index}
+      value={option}
+      as={motion.div}
+      ref={ref}
+      initial="hidden"
+      animate={inView ? "visible" : "hidden"}
+      variants={itemVariants}
+      className="group flex select-none items-center gap-2 rounded-lg p-2 data-[selected]:bg-primary/10"
```

Comment:
I think it would look good if the select options had some visual feedback when being hovered, maybe something like this:
```suggestion
      className="group flex select-none items-center gap-2 rounded-lg p-2 data-[selected]:bg-primary/10 cursor-pointer hover:bg-primary/10 transition-colors"
```